site/docs/windows.md

```diff
@@ -137,6 +137,8 @@ To build C++ targets with MSVC, you need:
 
 *   [The Visual C++ compiler](install-windows.html#install-vc).
 
+*   The compiler flag `/showIncludes` in case that you define your custom toolchain.
```

Comment:
This does not really help users. A pointer to how to set up a custom MSVC toolchain might be useful.  We don't want users to add /showIncludes, you want the toolchain author to set it up correctly. This hints that it is the user's problem.

Reply:
Agree, normal users should only need to know this when they are trying to write their own custom MSVC toolchain. Adding this as a general requirement doesn't sound right.

Reply:
> This does not really help users. A pointer to how to set up a custom MSVC toolchain might be useful. We don't want users to add /showIncludes, you want the toolchain author to set it up correctly. This hints that it is the user's problem.

@aiuto I understand your point, but everything depends on who you understand as a user. From your comment I get that this documentation is from your point of view intended for the end-user, the average software developer that do not have to create a toolchain. However in the same documentation page, in the section of clang is talking about the need of registering a platform, thing that I would also not expect an average user to have to do, but the person that sets up the Bazel. https://docs.bazel.build/versions/master/windows.html#build-c-with-clang

When you talk about a pointer to how to set up a custom MSVC toolchain, is there something already? Or you mean a pointer plus create the page explaining it?

> Agree, normal users should only need to know this when they are trying to write their own custom MSVC toolchain. Adding this as a general requirement doesn't sound right.

@meteorcloudy can we keep the ticket that I created https://github.com/bazelbuild/bazel/issues/13220 to improve the documentation or should I create a new one? Not being mentioned in the documentation and not having an open ticket for that, will only make people struggle on the same issue that I had for hours. I understand that for a Bazel developer is expected this behavior but other user (including toolchain writers) not involved with Bazel development might find this behavior hard to predict.

@meteorcloudy @aiuto in general the idea why I push for this transparency is because right now it is not easy to create a toolchain for MSVC similar to the autogenerated one https://github.com/bazelbuild/bazel/issues/12767 and when trying to do that I spent several hours trying to understand why it is failing when not specifying showIncludes but passing otherwise. I would like to prevent more people spending time like me on something that could be documented.

Reply:
@limdor Thanks! I totally understand. I changed the original issue to reflect documentation issue. Please use that to track. I also suggested a place where this should be documented.

As for creating a toolchain for MSVC, maybe https://github.com/meteorcloudy/my_tests/tree/master/preconfigure_msvc_toolchain can help? Instead of writing one entirely from scratch, maybe you can start with the auto generated one?